core/worker.go

```diff
@@ -736,15 +736,36 @@ func (w *worker) commitUncle(env *environment, uncle *types.WorkObjectHeader) er
 func (w *worker) commitTransaction(env *environment, parent *types.WorkObject, tx *types.Transaction) ([]*types.Log, error) {
 	if tx != nil {
 		if tx.Type() == types.ExternalTxType && tx.To().IsInQiLedgerScope() {
-			if err := env.gasPool.SubGas(params.CallValueTransferGas); err != nil {
-				return nil, err
-			}
-			if tx.Value().Int64() > types.MaxDenomination {
-				return nil, fmt.Errorf("tx %032x emits UTXO with value greater than max denomination", tx.Hash())
+			gasUsed := env.wo.GasUsed()
+			txGas := tx.Gas()
+			if tx.ETXSender().Location().Equal(*tx.To().Location()) { // Quai->Qi conversion
+				lock := new(big.Int).Add(env.wo.Number(w.hc.NodeCtx()), big.NewInt(params.ConversionLockPeriod))
+				value := misc.QuaiToQi(env.wo, tx.Value())
+				denominations := misc.FindMinDenominations(value)
+				for i, denomination := range denominations { // TODO: Decide maximum number of iterations
+					if txGas < params.CallValueTransferGas {
```

Comment:
does this mean that the bigger denominations go in first in the iteration?

Reply:
Yes that's correct

Reply:
Actually this is a good point, it has to be in order of denomination. Updated to reflect this